docs/configuration.asciidoc

```diff
@@ -972,31 +972,19 @@ The `sanitizeFieldNames` configuration value allows you to configure a list of w
 case-insensitive by default. You may make wildcard searches case-sensitive by
 using the `(?-i)` prefix. These patterns apply to the request and response HTTP headers, as well as any form field captured during an `application/x-www-form-urlencoded` data request.
 
-The `sanitizeFieldNames` will redact any matched _field names_.  If you wish to filter or _redact_ other data the <<filter-http-headers,`filterHttpHeaders`>> configuration field or the <<apm-add-filter,API filtering functions>> may be a better choice.
+The `sanitizeFieldNames` will redact any matched _field names_.  If you wish to filter or _redact_ other data the <<apm-add-filter,API filtering functions>> may be a better choice.
 
 [[filter-http-headers]]
 ==== `filterHttpHeaders`
 
+[small]#Deprecated in: v3.x.x
+
 * *Type:* Boolean
 * *Default:* `true`
 * *Env:* `ELASTIC_APM_FILTER_HTTP_HEADERS`
 
-When tracing an incoming HTTP request,
-the agent will add metadata about the requests to the recorded transaction.
-The same applies for errors that occur and are captured as a result of the request.
-
-This boolean specifies if the agent should anonymize certain sensitive HTTP headers by default before they are sent to the APM Server.
-When anonymized,
-the header value will be set to `[REDACTED]`
-
-Currently, the following HTTP headers are anonymized by default:
-
-* `Authorization` - The full value of this header is redacted.  In versions of the agent greater than v3.9, the authorization header is redacted by the default configuration of <<sanitize-field-names>>.
-* `Cookie` - The cookies inside the `Cookie` header are analyzed and their values redacted if they appear sensitive (like a session cookie).
-  See the https://github.com/watson/is-secret[is-secret] module for details about which patterns are considered sensitive.
-
-If you wish to filter or sanitize other data,
-use one of the <<apm-add-filter,filtering>> functions.
+This configuration is *deprecated* and will be removed in a future major version. If you whish to filter or or _redact_ other 
+data the <<apm-add-filter,API filtering functions>> or <<sanitize-field-names, Sanitize fields configuration>> may be a better choice.
```

Comment:
It would be helpful for this description to explain why this is deprecated and what a user using this config var should replace it with.  In this case `filterHttpHeaders: true` can just be dropped because it is the default behaviour, and `filterHttpHeaders: false` can be mostly replicated by changing the values in `sanitizeFieldNames` (with the caveat that the parsed cookie values are now on `transaction.request.cookies` rather than `transaction.request.headers.cookie`).  Do I have that right?

Reply:
sanitization of the cookie string was actually using `sanitizeFieldNames` to decide which cookie needs to be redacted. So your assumption is correct. We can provide a value for `sanitizeFieldNames` which will be the equivalent of `filterHttpHeaders: false`